src/Cache.js

```diff
@@ -71,22 +71,21 @@ class Cache {
    */
   expireAfter(key: any) {
     const {maxAge, onExpire} = this.options;
+    const indexOfKey = findExpirationIndex(this.expirations, key);
 
     const timeoutId = setTimeout(() => {
-      this.remove(key);
-      this.expirations.splice(findExpirationIndex(this.expirations, key), 1);
-
       if (isFunction(onExpire)) {
         // $FlowIgnore onExpire is a function
-        onExpire(key.key);
+        if (onExpire(key.key) === false) {
+          return this.expireAfter(key);
+        }
       }
+      this.remove(key);
+      this.expirations.splice(indexOfKey, 1);
```

Comment:
so there is a reason I was calling `findExpirationIndex` twice, once for the clearance and once inside the method, because there is the possibility that the indices will not align (`setTimeout` order of firing is not guaranteed). please return the additional call inside the `splice`, as this guarantees the contract.

also, there is no need for the nested `if` statements.

```javascript
if (isFunction(onExpire) && onExpire(key.key) === false) {
  return this.expireAfter(key);
}

this.remove(key);
this.expirations.splice(findExpirationIndex(this.expirations, key), 1);
```